homeassistant/components/sensor/wunderground.py

```diff
@@ -657,21 +654,63 @@ def __init__(self, rest, condition):
         """Initialize the sensor."""
         self.rest = rest
         self._condition = condition
+        self._state = None
+        self._attributes = {
+            ATTR_ATTRIBUTION: CONF_ATTRIBUTION,
+        }
+        self._icon = None
+        self._entity_picture = None
+        self._unit_of_measurement = self._cfg_expand("unit_of_measurement")
         self.rest.request_feature(SENSOR_TYPES[condition].feature)
+        self._long_text = None
 
     def _cfg_expand(self, what, default=None):
+        """Parse and return sensor data."""
         cfg = SENSOR_TYPES[self._condition]
         val = getattr(cfg, what)
-        try:
-            val = val(self.rest)
-        except (KeyError, IndexError) as err:
-            _LOGGER.warning("Failed to parse response from WU API: %s", err)
-            val = default
-        except TypeError:
-            pass  # val was not callable - keep original value
+        if callable(val):
```

Comment:
Let's switch to a guard pattern here so we can prevent extra indentation:
```python
if not callable(val):
    return val
try:
    …
```

Reply:
Changed